addons/mail/models/mail_activity.py

```diff
@@ -588,7 +588,51 @@ def get_activity_data_format(self, activity_ids, message_ids=None):
         }
 
     @api.model
-    def get_activity_data(self, res_model, domain, limit=None, offset=0):
+    def get_activity_data(self, res_model, domain, limit=None, offset=0,
+                          activity_filters=None, activity_search_fields=None):
+        """ Get records along with their activity data according to the search criterion.
+
+        The records to fetch is controlled by the parameters: res_model, domain, limit
+        and offset. If the activity_filters is None, all the activities related to the
+        records are retrieved otherwise only the activities that also match those filters
+        are retrieved.
+        Note that the filters: activities_state_overdue, activities_state_today,
+        activities_state_planned are combined with an OR to match how they are used in
+        the interface (they are grouped in the same section).
+
+        :param str res_model: model of the records to fetch
+        :param list domain: record search domain
+        :param int limit: maximum number of records to fetch
+        :param int offset: offset of the first record to fetch
+        :param list|None activity_filters: list of activated filters among activities_my,
+            activities_state_overdue, activities_state_today, activities_state_planned.
+        :param list|None activity_search_fields: list of search on field to applies on
+            activities. Each item is a couple defining the field followed by the value
+            to search.
+            Supported fields: activity_user_id, activity_create_uid, activity_state
+        """
+        user_tz = self.user_id.sudo().tz
+
+        # 1. Retrieve all ongoing and completed activities according to the parameters
+        activity_filters = set(activity_filters) if activity_filters else set()
+        activity_filter_states = {filter_name[len('activities_state_'):]
+                                  for filter_name in activity_filters if filter_name.startswith('activities_state_')}
+        activity_search_fields = dict(activity_search_fields) if activity_search_fields else dict()
```

Comment:
Seems we support only a limited, so maybe could be made explicit like
```suggestion
        search_terms = {
            fname: fvalue
            for fname, fvalue in (activity_search_fields or [])
            if fname in {'activity_user_id', 'activity_create_uid', 'activity_state'}
        }
```